src/core/manager.c

```diff
@@ -760,7 +760,7 @@ int manager_new(UnitFileScope scope, unsigned test_run_flags, Manager **_m) {
                 goto fail;
 
         r = manager_setup_cgroup(m);
-        if (r < 0)
+        if (r < 0 && test_run_flags == 0)
```

Comment:
It seems that `systemd --test` has broken since 10bd3e2e4c21bbf74480e686b9de564a9f6d0a4e was merged. Having skimmed over the commit message and the code, I think that `manager_setup_cgroup` should be fixed instead.

Reply:
Am I correct in thinking that https://github.com/systemd/systemd/issues/8072 is exactly the same issue as this one?

Reply:
Yes, it seems to be the same issue.

Reply:
Looking at the old code, it used to have the condition ```if (!m->test_run_flag)``` and skipped good chunk of the ```manager_setup_cgroup()``` when ```--test``` mode is enabled.

Either way, we need to test for ```test_run_flag``` somewhere. I will prepare alternative version which pushes the check back into ```manager_setup_cgroup()```.

I figure I should also add a test for this so we don't accidentally break the test mode again in the future.

Reply:
> Either way, we need to test for test_run_flag somewhere. I will prepare alternative version which pushes the check back into manager_setup_cgroup().

On a second thought, I think that current approach is better. Closer the ```--test``` mode execution to the real thing the better. So trying to setup cgroups and gracefully handling failure in test mode is better than basically skipping the function. Also, cases when you don't have session scope and hence you will be unable to actually create sub-cgroup are pretty rare during normal development.

Reply:
I agree that the closer to the reality the test mode is, the better, but `manager_setup_cgroup` seems to be a more convenient place to decide whether an error should be ignored or not. I also find this `if` a bit incosistent because all the other `manager_setup` functions seem to be able to deal with the test mode without any support from outside.

Reply:
You are right, consistency is important. Newest version handles the error in ```manager_setup_cgroup()```.